central/declarativeconfig/manager_impl.go

```diff
@@ -0,0 +1,41 @@
+package declarativeconfig
+
+import (
+	"context"
+	"time"
+
+	"github.com/stackrox/rox/pkg/k8scfgwatch"
+	"github.com/stackrox/rox/pkg/logging"
+	"github.com/stackrox/rox/pkg/sync"
+)
+
+const (
+	watchInterval        = 5 * time.Second
+	declarativeConfigDir = "/run/stackrox.io/declarative-configuration"
+)
+
+var (
+	log = logging.LoggerForModule()
+)
+
+type managerImpl struct {
+	once sync.Once
+}
+
+// New creates a new instance of Manager.
+// Note that it will not watch the declarative configuration directories when creaeted, only after
+// WatchDeclarativeConfigDir has been called.
+func New() Manager {
+	return &managerImpl{}
+}
+
+func (m *managerImpl) WatchDeclarativeConfigDir() {
+	m.once.Do(func() {
+		wh := &watchHandler{m: m}
+		// Set Force to true, so we explicitly retry watching the files within the directory and not stop on the first
+		// error occurred.
+		watchOpts := k8scfgwatch.Options{Interval: watchInterval, Force: true}
+		_ = k8scfgwatch.WatchConfigMountDir(context.Background(), declarativeConfigDir,
```

Comment:
Shouldn't we at least log this error?

Reply:
When `force == true`, no error will be returned by [WatchConfigMountDir](https://github.com/stackrox/stackrox/blob/2bd26143bfbe8de03bd59e9f3bb4a00b264383c0/pkg/k8scfgwatch/watch.go#L36-L40). Instead, the error will be passed to the watch handler's `OnWatchError`. I've added a log there.